### ISSUE TITLE
Plotter closes automatically

### ISSUE THREAD

**Issue by laserman781** [pyvista/pyvista-support#53]
Every time I run plot something with Plotter, the first time it opens fine, but any other time it closes automatically without fail.

**Comment by banesullivan** [comment]
Can you provide code to reproduce this?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by laserman781** [comment]
This is the code producing the error:
I tired restarting the kernal and the same thing happens every time...
```
@interact_calc(holes='Enter file path', blocks='Enter file path')
def selection(holes, blocks):
    with out:
        clear_output()
        if blocks != 'Enter file path':
            grid = pv.read(blocks+'.vtk')
        H = pv.read(holes+'.vtk')
        h = widgets.SelectMultiple(options=DH['Name'],description='Drillholes',disabled=False)
        print('Specific Query')
    @interact_calc(select=dh,regex='Enter selection')
    def plot(select,regex):
        with out:
            clear_output()
            if regex != 'Enter selection':
                extracted = query(H,'Name',regex)
            else:
                extracted = query(H,'Name',select)
            print(extracted)
        @interact_calc(savename = 'Enter extracted drillholes name')
        def plot(savename):
            with out:
                clear_output()
                extracted.save(savename+'.vtk')
                print('saved')
                p = pv.Plotter() 
                surface = extracted.extract_surface()  # Extract the outer surface
                cells = surface.cell_centers()   # Grabb the cell centers
                p.add_mesh(extracted, show_edges=True)
                p.add_point_labels(cells, cells.active_scalar_name)
                p.remove_scalar_bar()
                if blocks != 'Enter file path':
                    p.add_mesh(grid)
                p.show()
```

**Comment by banesullivan** [comment]
I have a feeling this has to do with the fact that you have two `plot` functions in your namespace... try renaming the second one.

**Comment by laserman781** [comment]
So I fixed this, and still the same issue...

**Comment by banesullivan** [comment]
I cannot reproduce anything from what you shared... can you provide a simplified, reproducible example?

Also, maybe screen captures of what you're experiencing? And a more detailed explanation of this phenomena...

**Comment by laserman781** [comment]
Here is a more simplified code where I receive the same issue...
I have attached a screen capture of the issue I am seeing. (I just realized)
![Plotter-Issue](https://user-images.githubusercontent.com/12982931/63521717-fb48d800-c4c4-11e9-9250-129745487470.gif)
```
import pyvista as pv
DH = pv.read('sample_drillholes.vtk')
p = pv.Plotter(notebook = False)
p.add_mesh(DH, show_edges=True)
p.remove_scalar_bar()
p.show()
```

**Comment by laserman781** [comment]
I just realized my gif didn't come out right...

**Comment by banesullivan** [comment]
Huh that’s very strange. Can you share a report from that same notebook:

```
import pyvista as pv
print(pv.Report())
```

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by laserman781** [comment]
See below:
--------------------------------------------------------------------------------
  Date: Thu Aug 22 10:40:16 2019 Eastern Daylight Time

           Windows : OS
                 4 : CPU(s)
             AMD64 : Machine
             64bit : Architecture
           11.7 GB : RAM
           Jupyter : Environment

  Python 3.7.3 (default, Apr 24 2019, 15:29:51) [MSC v.1915 64 bit (AMD64)]

            0.22.0 : pyvista
             8.1.2 : vtk
            1.16.4 : numpy
             2.5.0 : imageio
             1.4.3 : appdirs
             0.4.3 : scooby
             3.1.0 : matplotlib
   Version unknown : PyQt5
             7.6.1 : IPython
             7.5.0 : ipywidgets
             1.0.0 : colorcet

  Intel(R) Math Kernel Library Version 2019.0.4 Product Build 20190411 for
  Intel(R) 64 architecture applications

**Comment by banesullivan** [comment]
And you noticed this only after upgrading to version `0.22.0`?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by laserman781** [comment]
This was never an issue until just recently...

**Comment by banesullivan** [comment]
Would you please also run the same code from a Python interpreter on the command line to see if it happens there too?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by laserman781** [comment]
![Plotter](https://user-images.githubusercontent.com/12982931/63525710-1f5be780-c4cc-11e9-9dfc-b3f00f271b73.gif)

It seems to work fine this way

**Comment by banesullivan** [comment]
That's good to hear it is at least working from the command line. 

I'll investigate a bit further to see if I can reproduce this - currently I have no idea why you are experiencing this issue in Jupyter notebooks.

In that same notebook where the issue appears, could you print the following:

```py
import pyvista as pv
print('OFF_SCREEN: ', pv.OFF_SCREEN)
```

**Comment by laserman781** [comment]
I receive this:
OFF_SCREEN:  False

I've tried creating new notebooks, reinstalling PyVista and I get the same issue every time

**Comment by banesullivan** [comment]
very, very strange. Try this snippet as well just in case...

```py
import pyvista as pv
DH = pv.read('sample_drillholes.vtk')
p = pv.Plotter(notebook=False, off_screen=False)
p.add_mesh(DH, show_edges=True)
p.remove_scalar_bar()
p.show()
```

**Comment by banesullivan** [comment]
And then try this.... with the `auto_close` flag

```py
import pyvista as pv
DH = pv.read('sample_drillholes.vtk')
p = pv.Plotter(notebook=False, off_screen=False)
p.add_mesh(DH, show_edges=True)
p.remove_scalar_bar()
p.show(auto_close=False)
```

**Comment by laserman781** [comment]
In the first one, the same things happen. The second one works fine, although if I run the first one, and then run the second one after it crashes and the kernel dies.

**Comment by banesullivan** [comment]
@laserman781, are you still experiencing this issue?

**Comment by rgangrade** [comment]
I was experiencing the same issue. Tried researching a bit and used '%matplotlib.qt' before the scene. Works fine for me. I am quite new to Python so this may not be a great solution, but it works fine!

**Comment by banesullivan** [comment]
Thanks, @rgangrade! That's a weird fix, so I'll have to take a look at what's going on there

**Comment by GuillaumeFavelier** [comment]
I'll close this since it has been taken care of by https://github.com/pyvista/pyvista/pull/502

The fix is now available in the [version 0.23.0 of pyvista](https://github.com/pyvista/pyvista/releases/tag/0.23.0) !
